src/clients/c++/CMakeLists.txt

```diff
@@ -30,23 +30,37 @@ cmake_minimum_required (VERSION 3.5)
 # librequest.so
 #
 find_package(CURL REQUIRED)
+
 add_library(
-  request SHARED
-  request.cc request.h
-  request_common.cc request_common.h
-  request_http.cc request_grpc.cc
-  $<TARGET_OBJECTS:model-config-library>
-  $<TARGET_OBJECTS:proto-library>
-  $<TARGET_OBJECTS:grpc-library>
-)
+    request SHARED
+    request.cc request.h
+    request_common.cc request_common.h
+    request_http.cc request_grpc.cc
+    $<TARGET_OBJECTS:model-config-library>
+    $<TARGET_OBJECTS:proto-library>
+    $<TARGET_OBJECTS:grpc-library>
+  )
+
+# use different syntax for Windows
+if(WIN32)
```

Comment:
You can list target_link_libraries multiple times, so can everything be common except -lpthread (use ${CURL_LIBRARY_RELEASE} for both??). Then add:

if(NOT WIN32)
target_link_libraries(
  request
  -lpthread
)